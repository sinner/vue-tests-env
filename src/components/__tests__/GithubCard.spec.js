import { mount } from '@vue/test-utils';
import GithubCard from '@/components/GithubCard.vue';

describe('GithubCard.vue', () => {
  it('should work properly the composeUrl method', () => {
    const wrapper = mount(GithubCard);

    const { vm } = wrapper;
    const { composeUrl } = GithubCard.methods;

    expect(composeUrl(123)).toBe('https://api.github.com/users/123');

    vm.$nextTick(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should work properly the fetchData method', async () => {
    const jsonMock = jest.fn().mockResolvedValue('Github Data');
    window.fetch = jest.fn().mockResolvedValue({
      json: jsonMock,
    });

    const wrapper = mount(GithubCard, {
      methods: {
        composeUrl: () => 'url',
      },
    });

    const { vm } = wrapper;

    await vm.fetchData();

    expect(window.fetch).toHaveBeenCalledWith('url');
    expect(jsonMock).toHaveBeenCalled();
    expect(vm.data).toBe('Github Data');

    // vm.$nextTick(() => {
    //   expect(vm.degrees).toBe(50);
    //   expect(vm.type).toBe('fahrenheit');
    //   expect(wrapper).toMatchSnapshot();
    // });
  });
});
