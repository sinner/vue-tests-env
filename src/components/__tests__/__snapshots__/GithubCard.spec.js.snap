// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubCard.vue should work properly the composeUrl method 1`] = `
<div>
  <form><input type="text"> <button type="submit">Load Details...</button></form>
  <!---->
</div>
`;
